Meadowlark-js/packages/meadowlark-core/src/helpers/DocumentId.ts

```diff
@@ -4,33 +4,36 @@
 // See the LICENSE and NOTICES files in the project root for more information.
 
 import JsSha from 'jssha';
-import invariant from 'invariant';
-import { EntityInfo } from '../model/EntityInfo';
+import { DocumentElement } from '../model/DocumentElement';
+import { DocumentIdentity } from '../model/DocumentIdentity';
+import { DocumentInfo } from '../model/DocumentInfo';
 
 /**
  * Returns a SHAKE128 hash of length 224 bits for the given naturalKeyString for use as a document id.
- * naturalKeyString must be of the form "NK#<hash-separated key-value pairs>"
  */
-export function documentIdFromNaturalKeyString(naturalKeyString: string): string {
-  invariant(naturalKeyString.startsWith('NK#'), `naturalKeyString "${naturalKeyString}" did not start with "NK#"`);
-  // Hack that might be better addressed elsewhere
-  const nks = naturalKeyString.replace(/\.school=/g, '.schoolId=');
+export function idFromDocumentElements(documentIdentity: DocumentIdentity): string {
+  // TODO: This needs to be investigated (see RND-234) Might be due to a problem with extracted document reference paths.
+  // const nks = documentIdentity.replace(/\.school=/g, '.schoolId=');
```

Comment:
Commented out this string hack-fix that I kind of remember. I think we are missing some logic around certain extracted document reference scenarios. Created RND-234 to follow up.